ocp_resources/network_attachment_definition.py

```diff
@@ -24,4 +46,149 @@ def to_dict(self):
                 f"{NamespacedResource.ApiGroup.K8S_V1_CNI_CNCF_IO}/resourceName": self.resource_name
             }
         res["spec"] = {}
+        if self.config:
+            res["spec"]["config"] = self.config
+        else:
```

Comment:
This could co-exist like this:
```python
res["spec"]["config"] = {}
if self.config:
    res["spec"]["config"] = self.config
if self.cni_version:
    res["spec"]["config"]["cniVersion"] = self.cni_version
if self.cni_type:
    res["spec"]["config"]["type"] = self.cni_type
```

Reply:
Ok, but that is not a nice API in general. You are asking the caller to know this, which is not that trivial.
But as you already do things like that (e.g. the input file), I guess you can call it "consistency".

Reply:
Added docstring.